instrumentation/opentelemetry_req/lib/opentelemetry_req.ex

```diff
@@ -30,39 +108,82 @@ defmodule OpentelemetryReq do
   )
   ```
 
-  ### Example without path_params
+  Request and response header attributes are opt-in and can be set with the
+  `request_header_attrs` and `response_header_attrss` options. Values should be lower-case.
+
+  ### Trace Header Propagation
 
+  By default, trace propagation headers are not injected to requests. There are
+  two options available to propagate trace headers:
+
+    * set `propagate_trace_headers` option to `true` when attaching or in the call
+    * manually use `:otel_propagator_text_map.inject/1`
+
+  Example:
   ```
   client =
     Req.new()
-    |> OpentelemetryReq.attach(
-      base_url: "http://localhost:4000",
-      propagate_trace_ctx: true,
-      no_path_params: true
-    )
+    |> OpentelemetryReq.attach(propagate_trace_headers: true)
 
-  client
-  |> Req.get(
-    url: "/api/users"
-  )
+  # or
+
+  client =
+    Req.new()
+    |> OpentelemetryReq.attach()
+
+  Req.get(client, "/", propagate_trace_headers: true)
   ```
-  If you don't set `path_params` the request will raise.
-  """
 
-  alias OpenTelemetry.Tracer
-  alias OpenTelemetry.SemanticConventions.Trace
-  require Trace
-  require Tracer
-  require Logger
+  ### Span Name Override
+
+  The span name can be overridden by setting the `span_name` option in the call.
+
+  Example:
+  ```
+  client =
+    Req.new()
+    |> OpentelemetryReq.attach()
 
+  Req.get(client, "/", span_name: "custom")
+  ```
+
+  > #### Option Precedence {: .info}
+  >
+  > Options passed in a request take precedence over those passed in `attach/1`.
+  >
+  """
+
+  @spec attach(Req.Request.t(), options()) :: Req.Request.t()
   def attach(%Req.Request{} = request, options \\ []) do
+    opts = NimbleOptions.validate!(options, @options_schema) |> Enum.into(%{})
+
+    config =
+      opts
+      |> Enum.into(%{})
```

Comment:
We don't need to duplicate the `Enum.into(%{})` in the config processing, but also maybe we could just combine it all into one pipe? It looks like nothing is using `opts` except the `Enum.member?` which seems like it could just as easily check for the value being in the `config` Map.

```suggestion
    config =
      options
      |> NimbleOptions.validate!(@options_schema)
      |> Map.new()
```